query/src/sql/sql_parser.rs

```diff
@@ -1045,8 +1045,7 @@ impl<'a> DfParser<'a> {
                 break;
             }
             let name = name.unwrap();
-            let eq = self.parser.expect_token(&Token::Eq);
-            if eq.is_err() {
+            if !self.parser.consume_token(&Token::Eq) {
                 // only paired values are considered as options
                 self.parser.prev_token();
```

Comment:
no need pre_token any more if using `consume_token`

Reply:
I am recover the  _name_ that already been consumed since it will not be an option if the next token is not a _=_.

Reply:
```
    fn consume_token(&mut self, expected: &str) -> bool {
        if self.parser.peek_token().to_string().to_uppercase() == *expected.to_uppercase() {
            self.parser.next_token();
            true
        } else {
            false
        }
    }
```

I do think it's buggy, if the next token is not a `=`, it will not call `next_token`.

`peek_token` will not advance the iterator.